### PR TITLE
feat/add flag to analyze button handler to prevent button spamming

### PR DESCRIPTION
**Pull request by frankie-mur**
#32 This PR adds a value `IsAnalyzeButtonActive` flag to the CriticWindow struct to prevent spamming of analyze button 

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the Analyze button to prevent multiple analyses from being initiated simultaneously.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -113,15 +113,25 @@ func onFileOpenButtonClickedHandler() {
 }
 
 func onAnalyzeButtonClickedHandler() {
+	// Prevent re-entry if already processing
+	if criticWindow.IsAnalyzeButtonActive {
+		return
```
[the remark this answers is not in the document]
Comment (frankie-mur) [reply main.go]: Good call! Will do

File: main.go [hunk 2]
```diff
@@ -113,15 +113,25 @@ func onFileOpenButtonClickedHandler() {
...
+	}
+
+	criticWindow.IsAnalyzeButtonActive = true
 	diff := criticWindow.DiffPanel.TextGrid.Text()
 
 	if diff == "" || criticWindow.DiffPanel.IsDefaultText() {
 		dialog.ShowError(fmt.Errorf("the diff is empty"), *criticWindow.Window)
+		criticWindow.IsAnalyzeButtonActive = false
```
[the remark this answers is not in the document]
Comment (frankie-mur) [reply main.go]: Hmm, great point I will update this

File: main.go [hunk 3]
```diff
@@ -113,15 +113,25 @@ func onFileOpenButtonClickedHandler() {
...
 
 		return
 	}
 
-	go getCodeReview(diff)
+	go func() {
+		getCodeReview(diff)
+		criticWindow.IsAnalyzeButtonActive = false
+	}()
```
[the remark this answers is not in the document]
Comment (frankie-mur) [reply main.go:136]: Thanks for the tip! It brings up a concern I have of updating the UI via this goroutine, is this an ideal approach? Would it be best to make these sequentially? What do you think?

[the remark this answers is not in the document]
Comment (frankie-mur) [reply main.go:141]: not quite, what I meant is that the getCodeReview() function itself includes code that updates the UI

[the remark this answers is not in the document]
Comment (kmesiab) [reply main.go]: This is great advice!  I will log a ticket for this in the future.

**Comment by kmesiab** [comment]
In the future, be sure to not merge commit, but rebase instead, to ensure a solid linear git history.

File: main.go [hunk 4]
```diff
@@ -115,6 +115,13 @@ func onFileOpenButtonClickedHandler() {
 }
 
 func onAnalyzeButtonClickedHandler() {
+	// Prevent re-entry if already processing
+	if criticWindow.IsAnalyzeButtonActive {
+		dialog.ShowInformation("Analyze in Progress", "Analysis is already in progress. Please wait until the current process is complete.", *criticWindow.Window)
```
Comment (kmesiab) [inline main.go:120]: Should we just disable the button?

Comment (frankie-mur) [reply main.go:120]: I tried this- unfortunately, I couldn't find a way to get the instance of the button from the `criticWindow.Toolbar` with the current implementation

Comment (kmesiab) [reply main.go:120]: Hmm, that is a design mistake on my part then. 

This is fine for now given that constraint and I'll plan for a refactor of the critic window struct.

**Comment by kmesiab** [comment]
Lgtm! Squash it and send it!

**Comment by frankie-mur** [comment]
Phew, done with squash, also Happy New Year's! 🎆

**Comment by kmesiab** [comment]
ref: https://github.com/kmesiab/ai-code-critic/issues/32

**Comment by kmesiab** [comment]
> Phew, done with squash, also Happy New Year's! 🎆

You too!

**Review by kmesiab — APPROVED** [review]

**Pull request merged**